src/pages/Stream/utils.ts

```diff
@@ -1,26 +1,25 @@
-import { Log } from "@/@types/parseable/api/query";
-import { LogStreamSchemaData } from "@/@types/parseable/api/stream";
-import { columnsToSkip } from "./providers/LogsProvider";
+import { Log } from '@/@types/parseable/api/query';
+import { LogStreamQueryWithFields } from '@/@types/parseable/api/stream';
+import { columnsToSkip } from './providers/LogsProvider';
 
 export const getPageSlice = (page = 1, perPage: number, data: Log[]) => {
 	const firstPageIndex = (page - 1) * perPage;
 	const lastPageIndex = firstPageIndex + perPage;
 	return data ? data.slice(firstPageIndex, lastPageIndex) : [];
 };
 
-export const makeHeadersFromSchema = (schema: LogStreamSchemaData | null): string[] => {
-	if (schema) {
-		const { fields } = schema;
-		return fields.map((field) => field.name);
-	} else {
-		return [];
+export const makeHeadersFromQueryFields = (queryResponse: LogStreamQueryWithFields | null): string[] => {
```

Comment:
`(queryResponse: LogStreamQueryWithFields | null)`
 
The `null` part should be present in the `LogStreamQueryWithFields` definition itself if you think the backend will respond with null at any point.